api/envoy/extensions/network/dns_resolver/apple/v3/apple_dns_resolver.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.extensions.network.dns_resolver.apple.v3;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.network.dns_resolver.apple.v3";
+option java_outer_classname = "AppleDnsResolverProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: apple DNS resolver]
+// [#extension: envoy.network.dns_resolver.apple]
+
+// For Envoy built in MacOS to use apple DNS library as DNS resolver, the following need to be done:
+// 1) In file: :repo:`extensions_build_config.bzl <source/extensions/extensions_build_config.bzl>`,
+// enable apple DNS resolver extension by adding below line
+// "envoy.network.dns_resolver.apple":         "//source/extensions/network/dns_resolver/apple:config",
+// 2) Either enable the run time flag: ``envoy.restart_features.use_apple_api_for_dns_lookups``, which
+// is enabled by default. Or in corresponding Envoy configuration, add a typed configuration. For example, in
```

Comment:
I updated the dynamic forward proxy configuration example with the c-ares DNS typed configuration in file: docs/root/configuration/http/http_filters/_include/dns-cache-circuit-breaker.yaml.
Here is the existing page: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/dynamic_forward_proxy_filter#config-http-filters-dynamic-forward-proxy

I didn't provide the apple DNS typed configuration example.  Adding it in the above page may not look nice also in Apple system, it will use APPLE DNS resolver by default anyway.